manifests/init.pp

```diff
@@ -45,6 +53,8 @@
   validate_bool($local_lib)
   validate_string($config_template)
   validate_string($package_ensure)
+  validate_string($ftp_proxy)
+  validate_string($http_proxy)
```

Comment:
I take that back.
> Note: validate_string(undef) will not fail in this version of the functions API.
https://github.com/puppetlabs/puppetlabs-stdlib#validate_string